indexer/scripts/elastic-conf.py

```diff
@@ -96,6 +116,59 @@ def main_loop(self) -> None:
             logger.error("One or more configurations failed. Check logs for details.")
             return
 
+    def repository_exists(self, es: Elasticsearch) -> bool:
+        repo_name = self.es_snapshot_repo
+        try:
+            response = es.snapshot.get_repository(name=repo_name)
+            return repo_name in response
+        except Exception:
+            logger.exception("Error validating repository: %s", repo_name)
+        return False
+
+    def register_repository(self, es: Elasticsearch) -> bool:
+        repo_type = self.es_snapshot_repo_type
+        if repo_type not in ["fs", "s3"]:
+            logger.error(
+                "Unsupported repository type: '%s'. Must be either 'fs' or 's3'",
+                repo_type,
+            )
+            return False
+
+        if self.repository_exists(es):
+            logger.info("%s repository already exists.", repo_type)
+            return True
+
+        if repo_type == "s3":
+            settings = {
+                "bucket": self.es_snapshot_s3_bucket,
+                "endpoint": self.es_snapshot_s3_endpoint,
+            }
+        elif repo_type == "fs":
+            settings = {"location": self.es_snapshot_fs_location}
+        else:
+            logger.error("Unsupported repository type: %s", repo_type)
```

Comment:
```suggestion
        else: # repo_type == "fs"
            settings = {"location": self.es_snapshot_fs_location}
```

We've already checked at the top of the method `if repo_type not in ["fs", "s3"]` so at this point it must be one of the two.